### PR TITLE
HiveBolt: misc code cleanups

### PR DESCRIPTION
**Pull request by dossett**
Contributing back very minor cleanups as I am trying to learn more about Storm code:
- javadoc fixes
- consolidate exception catches
- suppress unavoidable "unchecked" warnings
- remove unnecessary type parameters from constructors where possible

There should not be any functional changes introduced

### CODE REVIEW

**Comment by harshach** [comment]
@dossett not sure why these changes necessary here

File: external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public void prepare(Map conf, TopologyContext topologyContext, OutputCollector c
                 }
             }
             this.collector = collector;
-            allWriters = new HashMap<HiveEndPoint,HiveWriter>();
```
Comment (harshach) [inline external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java]: @dossett I feel like older one is much more readable than your change.

File: external/storm-hive/src/main/java/org/apache/storm/hive/common/HiveWriter.java [hunk 2]
```diff
@@ -249,9 +237,7 @@ public TransactionBatch call() throws Exception {
             });
         LOG.debug("Acquired {}. Switching to first txn", batch);
         batch.beginNextTransaction();
-        } catch(TimeoutException e) {
-            throw new TxnBatchFailure(endPoint, e);
-        } catch(StreamingException e) {
+        } catch(TimeoutException | StreamingException e) {
```
Comment (lazyval) [inline external/storm-hive/src/main/java/org/apache/storm/hive/common/HiveWriter.java]: This is java 7 syntax, isn't it? What's the policy regarding java 6 support?

File: external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java [hunk 3]
```diff
@@ -262,7 +261,7 @@ private void retireEldestWriter() {
     private int retireIdleWriters() {
         int count = 0;
         long now = System.currentTimeMillis();
-        ArrayList<HiveEndPoint> retirees = new ArrayList<HiveEndPoint>();
+        ArrayList<HiveEndPoint> retirees = new ArrayList<>();
```
Comment (lazyval) [inline external/storm-hive/src/main/java/org/apache/storm/hive/bolt/HiveBolt.java]: Diamond types are also Java 7 feature.

**Comment by dossett** [comment]
Adding anything specific to Java 7 was way beyond the scope of what I was trying to do here.  I will close the PR and perhaps open a new request with much more limited changes.  Thank you for the feedback everyone!

**Pull request closed without merging**